lib/src/main/java/org/team1540/base/util/robots/PIDTuningRobot.java

```diff
@@ -0,0 +1,164 @@
+package org.team1540.base.util.robots;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import edu.wpi.first.wpilibj.IterativeRobot;
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.command.Scheduler;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import org.team1540.base.Utilities;
+import org.team1540.base.preferencemanager.Preference;
+import org.team1540.base.preferencemanager.PreferenceManager;
+import org.team1540.base.util.SimpleCommand;
+import org.team1540.base.wrappers.ChickenTalon;
+
+/**
+ * Robot class to tune a Motion Magic PID controller.
+ *
+ * To change the motors to be tuned, change the preference values and then restart the robot code to
+ * allow the values to take effect. To disable a motor, set its motor ID to -1. Motor 1 will be
+ * configured as the master Talon and motors 2, 3, and 4 will be slaved to it in follower mode.
+ */
+public class PIDTuningRobot extends IterativeRobot {
+
+  @Preference(value = "P", persistent = false)
+  public double p;
+  @Preference(value = "I", persistent = false)
+  public double i;
+  @Preference(value = "D", persistent = false)
+  public double d;
+  @Preference(value = "F", persistent = false)
+  public double f;
+  @Preference(value = "I-Zone", persistent = false)
+  public int iZone;
+  @Preference(value = "Max I-Accum", persistent = false)
+  public int maxIAccum;
+  @Preference(value = "Max Acceleration", persistent = false)
+  public int maxAccel;
+  @Preference(value = "Max Velocity", persistent = false)
+  public int maxVel;
+  @Preference(value = "Setpoint", persistent = false)
+  public double setpoint;
+  @Preference(value = "Invert Sensor", persistent = false)
+  public boolean invertSensor;
+  @Preference(value = "Invert Output", persistent = false)
+  public boolean invertOutput;
+  @Preference(value = "Motor 1 ID", persistent = false)
```

Comment:
Is there some way of doing mass annotations in a way that'd allow for the `public boolean a, b`?

Reply:
No, and also we still need to supply labels. There is a `@TuningClass` annotation that puts every public field in a given class up but that's meant for, well, tuning classes.